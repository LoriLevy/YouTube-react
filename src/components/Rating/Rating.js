/* file: src/components/Rating/Rating.js */
import React from "react";
import "./Rating.scss";
import { Icon, Progress } from "semantic-ui-react";
import { formatNumberWithCommas } from "../../utilities/format-number";
function Rating(props) {
  let progress = null;
  let likeCount = 0;
  let dislikeCount = 0;
  if (props.likeCount && props.dislikeCount) {
    likeCount = formatNumberWithCommas(props.likeCount);
    dislikeCount = formatNumberWithCommas(props.dislikeCount);
    const percent =
      100 * (props.likeCount / (props.likeCount + props.dislikeCount));
    progress = <Progress className="progress" percent={percent} size="tiny" />;
  }
  return (
    <div className="rating">
      <div className="thumbs-up">
        <Icon name="thumbs outline up" />
        <span>{likeCount}</span>
      </div>
      <div className="thumbs-down">
        <Icon name="thumbs outline down" />
        <span>{dislikeCount}</span>
      </div>
      {progress}
    </div>
  );
}
export default Rating;
