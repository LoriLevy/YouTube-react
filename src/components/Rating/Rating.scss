.rating { /* root element */
    display: inline-grid;
    grid: auto auto / max-content max-content;  /*  max-content max-content means cram the entire content into the column without introducing additional line-breaks  */
    column-gap: 16px;
    grid-row-gap: 4px;

    .thumbs-up {
        grid-column: 1 / 2;
        grid-row: 1 / 2;
    }

    .thumbs-down {
        grid-column: 2 / 3;
        grid-row: 1 / 2;   /* Same row as thumbs-up */
    }

    .progress {
        grid-column: 1 / span 2;  /* span 2 columns  */
        grid-row: 2 / 3;   /*  row below thumbs-up/thumbs-down  */

        &.ui.progress:last-child {
            margin-bottom: 0;   /* Remove the margin that the Semantic progress bar has by default */
        }
    }
}