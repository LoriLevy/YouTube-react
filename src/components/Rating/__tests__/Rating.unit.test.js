import React from 'react';
import {shallow} from 'enzyme';
import Rating from '../Rating';

describe('No props, so see if Rating', () => {
    test('renders', () => {
        const wrapper = shallow(<Rating />);
        expect(wrapper).toMatchSnapshot();
    });
});
