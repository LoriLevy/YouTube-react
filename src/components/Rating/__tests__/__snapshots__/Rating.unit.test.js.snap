// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No props, so see if Rating renders 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span>
      0
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span>
      0
    </span>
  </div>
</div>
`;
